share/spack/setup-env.fish

```diff
@@ -0,0 +1,826 @@
+            else if test "x$elt" = "x--dependencies"
+                set __sp_subcommand_args $__sp_subcommand_args $elt
+            else
+                set __sp_module_args $__sp_module_args $elt
+            end
+
+        else
+            # bash compatibility: stop when the match first fails. Upon failure,
+            # we pack the remainder of `argv` into a global `__sp_remaining_args`
+            # array (`i` tracks the index of the next element).
+            set __sp_remaining_args (stream_args $argv[$i..-1])
+            return
+        end
+
+        # increment argument counter: used in place of bash's `shift` command
+        set -l i (math $i+1)
+
+    end
+
+    # if all elements in `argv` are matched, make sure that `__sp_remaining_args`
+    # is deleted (this might be overkill...).
+    set -e __sp_remaining_args
+end
+
+
+
+
+#
+# CHECK_SP_FLAGS, CONTAINS_HELP_FLAGS, CHECK_ENV_ACTIVATE_FLAGS, and 
+# CHECK_ENV_DEACTIVATE_FLAGS: support functions for checking arguments and flags.
+#
+
+function check_sp_flags -d "check spack flags for h/V flags"
+    #
+    # Check if inputs contain h or V flags.
+    #
+
+    # combine argument array into single string (space seperated), to be passed
+    # to regular expression matching (`string match -r`)
+    set -l _a "$argv"
+
+    # skip if called with blank input. Notes: [1] (cf. EOF)
+    if test -n "$_a"
+        if echo $_a | string match -r -q ".*h.*"
+            return 0
+        end
+        if echo $_a | string match -r -q ".*V.*"
+            return 0
+        end
+    end
+
+    return 1
+end
+
+
+function contains_help_flags -d "checks for help (-h/--help) flags"
+    #
+    # Check if inputs contain -h or --help
+    #
+
+    # combine argument array into single string (space seperated), to be passed
+    # to regular expression matching (`string match -r`)
+    set -l _a "$argv"
+
+    # skip if called with blank input. Notes: [1] (cf. EOF)
+    if test -n "$_a"
+        # looks for a single `-h` (possibly surrounded by spaces)
+        if echo $_a | string match -r -q " *-h *"
+            return 0
+        end
+
+        # looks for a single `--help` (possibly surrounded by spaces)
+        if echo $_a | string match -r -q " *--help *"
+            return 0
+        end
+    end
+
+    return 1
+end
+
+
+function check_env_activate_flags -d "check spack env subcommand flags for -h, --sh, --csh, or --fish"
+    #
+    # Check if inputs contain -h, --sh, --csh, or --fish
+    #
+
+    # combine argument array into single string (space seperated), to be passed
+    # to regular expression matching (`string match -r`)
+    set -l _a "$argv"
+
+    # skip if called with blank input. Notes: [1] (cf. EOF)
+    if test -n "$_a"
+        # looks for a single `-h` (possibly surrounded by spaces)
+        if echo $_a | string match -r -q " *-h *"
+            return 0
+        end
+
+        # TODO: should this crash (we're clearly using fish, not bash, here)?
```

Comment:
I think it shouldn't crash. For example, someone on bash may want to test the csh and fish output.

You also need to update `setup-env.sh` and `csh/spack.csh` to handle the new `--fish` flags, correct?

Reply:
Got it! I removed the todos